### ISSUE TITLE
OpenCL constant buffer size off by one

### ISSUE THREAD

**Issue by kgnk** [halide/Halide#1903]
Generated OpenCL code contains:

```
#if 4 < MAX_CONSTANT_BUFFER_SIZE && 0 < MAX_CONSTANT_ARGS
```

This will fail if the buffer size is 4. This should read:

```
#if 4 <= MAX_CONSTANT_BUFFER_SIZE && 0 < MAX_CONSTANT_ARGS
```

**Comment by kgnk** [comment]
I forgot to mention the constant data in this case is 4 bytes. This issue is more relevant if you have 32k or 64k of constant data and the limit is the same size.

**Comment by kgnk** [comment]
Fixed by #2014
